### PR TITLE
Fix to CAR 735528 Scan from String function displays wrong result if string exceeds 255 elements

### PR DESCRIPTION
**Pull request by cglzaguilar**
After some discussions, we decided on actually copying the whole buffer to guarantee numbers are correctly parsed.

### CODE REVIEW

File: test-it/ViaTests/ScanFromLongString.via [hunk 1]
```diff
@@ -0,0 +1,284 @@
+// Autogenerated Vireo assembly file
```
Comment (cglzaguilar) [inline test-it/ViaTests/ScanFromLongString.via]: WebVI is basically calling the following code:

Remaining

NotRemaining

File: source/include/DataTypes.h [hunk 2]
```diff
@@ -353,5 +353,96 @@ class FixedCArray : public SubVector<T>
...
+
+    //! Return the maximum capacity of the array.
+    IntIndex Capacity() { return _maxSize; }
+
+    //! Return a reference to the indexed element in the vector (no range checking).
+    const T&  operator[] (const int i) { return _buffer[i]; }
+
+    //! Append an element to the array if there is room.
+    Boolean Append(T element) {
+        IntIndex i = this->Length();
```
Comment (sanmut) [inline source/include/DataTypes.h]: nitpick: I would rename "i" to "length" or something. "i" is typically used as index.

File: source/include/DataTypes.h [hunk 3]
```diff
@@ -353,5 +353,96 @@ class FixedCArray : public SubVector<T>
...
+        if (i <= _maxSize) {
```
Comment (sanmut) [inline source/include/DataTypes.h]: I would replace _maxSize with Capacity() for clarity.

File: source/include/DataTypes.h [hunk 4]
```diff
@@ -353,5 +353,96 @@ class FixedCArray : public SubVector<T>
...
+            _buffer[i] = element;
+            this->_end++;
+            _buffer[this->Length()] = (T)0;
```
Comment (sanmut) [inline source/include/DataTypes.h]: As done in other places, I would replace this line "_buffer[this->Length()] = (T)0;" with *NonConstEnd() = (T)0;"

File: source/include/DataTypes.h (hunk 2 above)
Comment (cglzaguilar) [reply source/include/DataTypes.h]: Done

File: source/include/DataTypes.h (hunk 3 above)
Comment (cglzaguilar) [reply source/include/DataTypes.h]: Done.

File: source/include/DataTypes.h (hunk 4 above)
Comment (cglzaguilar) [reply source/include/DataTypes.h]: Done.

**Pull request merged**